### PR TITLE
Fix resolve logic in SolidMechanicsLagrangianFEM

### PR DESCRIPTION
**Pull request by paveltomin**
Wrong behavior for surfaceGeneration/DryFrac_StaticPenny_PrismElem.xml:
```
Rank 0: Split edge 5115 into edges 5115 and 22973
Rank 0: Split edge 5116 into edges 5116 and 22974
Rank 0: Split face 6158 into faces 6158 and 27184
Fracture Occurred. Resolve
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rsolid ) = ( 5.28e+00 )        ( R ) = ( 5.28e+00 )
        lagSolve: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rsolid ) = ( 5.34e-13 )        ( R ) = ( 5.34e-13 )
        Last LinSolve(iter,res) = (   1, 3.65e-15 ) ; 
Fracture Occurred. Resolve
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rsolid ) = ( 5.34e-13 )        ( R ) = ( 5.34e-13 )
        lagSolve: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rsolid ) = ( 4.55e-13 )        ( R ) = ( 4.55e-13 )
        Last LinSolve(iter,res) = (   1, 5.42e-16 ) ; 
Fracture Occurred. Resolve
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rsolid ) = ( 4.55e-13 )        ( R ) = ( 4.55e-13 )
        lagSolve: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rsolid ) = ( 4.33e-13 )        ( R ) = ( 4.33e-13 )
        Last LinSolve(iter,res) = (   1, 4.04e-16 ) ; 
Fracture Occurred. Resolve
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rsolid ) = ( 4.33e-13 )        ( R ) = ( 4.33e-13 )
        lagSolve: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rsolid ) = ( 4.23e-13 )        ( R ) = ( 4.23e-13 )
        Last LinSolve(iter,res) = (   1, 4.01e-16 ) ; 
Fracture Occurred. Resolve
```

After fix:
```
Rank 0: Split edge 5115 into edges 5115 and 22973
Rank 0: Split edge 5116 into edges 5116 and 22974
Rank 0: Split face 6158 into faces 6158 and 27184
Fracture Occurred. Resolve
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rsolid ) = ( 5.28e+00 )        ( R ) = ( 5.28e+00 )
        lagSolve: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rsolid ) = ( 5.34e-13 )        ( R ) = ( 5.34e-13 )
        Last LinSolve(iter,res) = (   1, 3.65e-15 ) ; 
lagSolve: Newton solver converged in less than 4 iterations, time-step required will be increased.
Rank 0: Writing out restart file at ./DryFrac_StaticPenny_PrismElem_restart_000000000/rank_0000000.hdf5
Time: 5.00e-01 s, dt: 0.5 s, Cycle: 1
```
In addition, add a safeguard check making sure no resolve occurred before exiting the timestep.

### CODE REVIEW

**Comment by paveltomin** [comment]
review anyone please?

File: src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp [hunk 1]
```diff
@@ -488,6 +489,7 @@ real64 SolidMechanicsLagrangianFEM::solverStep( real64 const & time_n,
 
       if( surfaceGenerator!=nullptr )
       {
+        locallyFractured = 0;
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp:492]:
```suggestion
       int  locallyFractured = 0;
       int globallyFractured = 0;
```

File: src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp [hunk 2]
```diff
@@ -469,8 +469,9 @@ real64 SolidMechanicsLagrangianFEM::solverStep( real64 const & time_n,
     int locallyFractured = 0;
     int globallyFractured = 0;
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp:470]:
```suggestion
    int locallyFractured = 0;
    int globallyFractured = 0;
```

**Review by CusiniM — APPROVED** [review]
Good catch. I think it will require a rebaseline

**Review by CusiniM — APPROVED** [review]
Since you are fixing this, we could define two members  
```c++
SolverBase * m_surfaceGenerator;
string m_surfaceGeneratorName;
```

and do this

```c++
`m_surfaceGenerator = &this->getParent().getGroup< SurfaceGenerator >( m_surfaceGeneratorName );`
```

in `postProcessInput` instead of having the name of the SufaceGeneartor hardcoded.

**Comment by paveltomin** [comment]
> Since you are fixing this, we could define two members
> 
> ```c++
> SolverBase * m_surfaceGenerator;
> string m_surfaceGeneratorName;
> ```
> 
> and do this
> 
> ```c++
> `m_surfaceGenerator = &this->getParent().getGroup< SurfaceGenerator >( m_surfaceGeneratorName );`
> ```
> 
> in `postProcessInput` instead of having the name of the SufaceGeneartor hardcoded.

Done, but didn't change phase field one that has the same hardcoded way.

**Pull request closed without merging**